java/daikon/inv/unary/string/dates/IsTimestampYYYYMMHHThhmmssmm.java

```diff
@@ -0,0 +1,99 @@
+package daikon.inv.unary.string.dates;
+
+import daikon.PptSlice;
+import daikon.inv.Invariant;
+import daikon.inv.InvariantStatus;
+import daikon.inv.OutputFormat;
+import daikon.inv.unary.string.SingleString;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.checkerframework.checker.lock.qual.GuardSatisfied;
+import org.checkerframework.dataflow.qual.Pure;
+import org.checkerframework.dataflow.qual.SideEffectFree;
+import typequals.prototype.qual.Prototype;
+
+/**
+ * Indicates that the value of a string variable is always a timestamp. Represented as x is
+ * Timestamp. Prints as {@code YYYY-MM-DDTHH:MM:SS.mmZ (Miliseconds are optional)}.
+ */
+public class IsTimestampYYYYMMHHThhmmssmm extends SingleString {
+  // We are Serializable, so we specify a version to allow changes to
+  // method signatures without breaking serialization.  If you add or
+  // remove fields, you should change this number to the current date.
+  static final long serialVersionUID = 20230704L;
+
+  // Variables starting with dkconfig_ should only be set via the
+  // daikon.config.Configuration interface.
+  /** Boolean. True iff Positive invariants should be considered. */
+  public static boolean dkconfig_enabled = false;
+
+  ///
+  /// Required methods
+  ///
+  private IsTimestampYYYYMMHHThhmmssmm(PptSlice ppt) {
+    super(ppt);
+  }
+
+  private @Prototype IsTimestampYYYYMMHHThhmmssmm() {
+    super();
+  }
+
+  private static @Prototype IsTimestampYYYYMMHHThhmmssmm proto =
+      new @Prototype IsTimestampYYYYMMHHThhmmssmm();
+
+  // Returns the prototype invariant
+  public static @Prototype IsTimestampYYYYMMHHThhmmssmm get_proto() {
+    return proto;
+  }
+
+  @Override
+  public boolean enabled() {
+    return dkconfig_enabled;
+  }
+
+  @Override
+  public IsTimestampYYYYMMHHThhmmssmm instantiate_dyn(
+      @Prototype IsTimestampYYYYMMHHThhmmssmm this, PptSlice slice) {
+    return new IsTimestampYYYYMMHHThhmmssmm(slice);
+  }
+
+  // A printed representation for user output
+  @SideEffectFree
+  @Override
+  public String format_using(
+      @GuardSatisfied IsTimestampYYYYMMHHThhmmssmm this, OutputFormat format) {
+    return var().name()
+        + " is Timestamp. Format: YYYY-MM-DDTHH:MM:SS.mmZ (Miliseconds are optional)";
+  }
+
+  @Override
+  public InvariantStatus check_modified(String v, int count) {
+    Pattern pattern =
```

Comment:
It is inefficient to re-compile this pattern every time that `check_modified()` is called.  Please store it in a static private variable instead.
The same comment apples to other regular expressions in this pull reqest.

Reply:
Fixed in https://github.com/codespecs/daikon/pull/497/commits/94c84606e997d498c302d61b21bceb48a7015c61